frank/utilities.py

```diff
@@ -856,6 +856,100 @@ def convolve_profile(r, I, disc_i, disc_pa, clean_beam,
     return I_smooth
 
 
+def add_vis_noise(vis, weights, seed=None):
+    r"""
+    Add Gaussian noise to visibilities
+
+    Parameters
+    ----------
+    vis : array, unit = [Jy]
+        Visibilities to add noise to.
+        Can be complex (real + imag * 1j) or purely real.
+    weights : array, unit = [Jy^-2]
+        Weights on the visibilities, of the form :math:`1 / \sigma^2`.
+        Injected noise will be scaled proportional to `\sigma`.
+    seed : int, default = None
+        Number to initialize a pseudorandom number generator for the noise draws
+
+    Returns
+    -------
+    vis_noisy : array, shape = (2, len(V)) if V is complex; else (1, len(V))
+        Visibilities with additive noise
+
+    """
+    if seed is not None:
+        np.random.seed(seed)
+
+    dim0 = 1
+    if np.iscomplexobj(vis):
+        dim0 = 2
+
+    noise = np.random.standard_normal([dim0, len(vis)])
```

Comment:
```suggestion
    vis = np.array(vis)
    noise = np.random.standard_normal((dim0,) + vis.shape)
```
This allows us to support arbitrary shapes for vis.